Sources/PackageModelSyntax/AddTarget.swift

```diff
@@ -414,3 +414,15 @@ fileprivate extension PackageDependency {
         )
     }
 }
+
+fileprivate extension TargetDescription {
+    var sanitizedName: String {
+        name
+            .spm_mangledToC99ExtendedIdentifier()
+            .capitalizingFirstLetter()
+    }
```

Comment:
Using the [Keyword enum ](https://github.com/swiftlang/swift-syntax/blob/main/Sources/SwiftSyntax/generated/Keyword.swift#L15) turned out to be not that trivial for this task, as there are many exceptions due to the keyword's context-specific nature, namely:
- `objc`, `escaping` and some others are considered non-valid identifiers only when they are used in combination with the leading `@` character.
- The same goes for `available` and `selector` but in combination with the `#` character
- `metadata` and `module` would also be valid identifiers, and I'm not even sure in which context they would be considered actual language keywords.
- etc.

As simple solutions are generally better, I preferred to use `c99name` in combination with a capitalization of the first letter. With the capitalization strategy, we can feed two birds with one scone: 
- It forces example type names to follow the overall type naming convention: "sanitizer" -> "struct Sanitizer"
- Reserved words are turned into valid identifiers when capitalized: "macro" -> "struct Macro"

I believe the combination of these two safeguards covers all known edge cases (to the best of my knowledge).

Reply:
Ah sorry, I missed the discussion above. We actually added `isValidSwiftIdentifier` in swift-syntax for the purpose you described above. Having said that, IMO `c99name` + uppercase is probably good enough here anyway.

Reply:
Ah, thanks! That's really nice to know.